src/sentry/integrations/messaging.py

```diff
@@ -199,3 +225,283 @@ def build_handler(
+            self.capture_metric("failure.get_identity")
+            return self._render_error_page(
+                request,
+                status=404,
+                body_text="HTTP 404: Could not find the identity.",
+            )
+
+        logger.info(
+            "get_identity_success",
+            extra={"integration_id": integration_id, "provider": self.provider_slug},
+        )
+        self.capture_metric("success.get_identity")
+        params.update({"organization": organization, "integration": integration, "idp": idp})
+
+        dispatch_kwargs = dict(
+            organization=organization, integration=integration, idp=idp, params=params
+        )
+        dispatch_kwargs = {k: v for (k, v) in dispatch_kwargs.items() if v is not None}
+        return super().dispatch(request, **dispatch_kwargs)
+
+    def get(self, request: Request, *args, **kwargs) -> HttpResponse:
+        params = kwargs["params"]
+        context = {"organization": params["organization"]}
+        integration = params.get("integration")
+        if integration:
+            context["provider"] = integration.get_provider()
+        return render_to_response(self.confirmation_template, request=request, context=context)
+
+    def post(self, request: Request, *args: Any, **kwargs: Any) -> HttpResponse:
+        if isinstance(request.user, AnonymousUser):
+            return HttpResponse(status=401)
+
+        try:
+            organization: RpcOrganization | None = kwargs.get("organization")
+            integration: Integration | None = kwargs.get("integration")
+            idp: IdentityProvider | None = kwargs.get("idp")
+
+            params_dict: Mapping[str, Any] = kwargs["params"]
+            external_id: str = params_dict[self.external_id_parameter]
+        except KeyError as e:
+            event = self.capture_metric("failure.post.missing_params", tags={"error": str(e)})
+            logger.exception(event)
+            return self._render_error_page(
+                request,
+                status=400,
+                body_text="HTTP 400: Missing required parameters.",
+            )
+
+        exc_response = self.persist_identity(idp, external_id, request)
+        if exc_response is not None:
+            return exc_response
+
+        self.notify_on_success(external_id, params_dict, integration)
+        self.capture_metric("success.post")
+        self.record_analytic("identity_linked", request.user.id)
+
+        if organization is not None:
+            self._send_nudge_notification(organization, request)
+
+        success_template, success_context = self.get_success_template_and_context(
+            params_dict, integration
+        )
+        return render_to_response(success_template, request=request, context=success_context)
+
+    def _send_nudge_notification(self, organization: RpcOrganization, request: Request):
+        # TODO: Delete this if no longer needed
```

Comment:
Is it not used?

Reply:
I don't think it is. Per the comment on line 422, I think this has been broken since the Hybrid Cloud split.

Reply:
This is not blocker for your PR since it was existing behavior but can you follow-up with the team in slack and figure out? Basically, is it important? -> add fix to backlog, if not -> delete